adapter.go

```diff
@@ -52,24 +52,26 @@ type CasbinRule struct {
 
 // Adapter represents the ETCD adapter for policy storage.
 type Adapter struct {
-	etcdEndpoints []string
-	key           string
+	etcdEndpoints         []string
+	key                   string
+	createKeyIfNotPresent bool
 
 	// etcd connection client
 	conn *client.Client
 }
 
-func NewAdapter(etcdEndpoints []string, key string) *Adapter {
-	return newAdapter(etcdEndpoints, key)
+func NewAdapter(etcdEndpoints []string, key string, createKeyIfNotPresent bool) *Adapter {
```

Comment:
Why are we accepting this option? I feel like it makes sense to always do it as it is an idempotent operation and the only place we are using it will set it to true.

Reply:
Makes sense to make this implicit.